### PR TITLE
refactor: angular api

### PR DESCRIPTION
**Pull request by quentinderoubaix**

### CODE REVIEW

File: angular/demo/src/app/samples/modal/playground.route.ts [hunk 1]
```diff
@@ -10,7 +10,8 @@ const undefinedConfig = getUndefinedValues(getModalDefaultConfig());
 	standalone: true,
 	imports: [AgnosUIAngularModule],
 	providers: provideHashConfig('modal'),
-	template: `<au-modal #widget></au-modal>`,
+	selector: 'app-playground-modal',
```
Comment (divdavem) [inline angular/demo/src/app/samples/modal/playground.route.ts:13]: As mentioned in #89 , please do not put a selector in the default component of `a .route.ts file`.

File: angular/demo/src/app/samples/rating/default.route.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import {Component} from '@angular/core';
 	standalone: true,
 	imports: [AgnosUIAngularModule],
 	template: `
-		<au-rating [(rating)]="rating" (hover)="hovered = $event" (leave)="left = $event" ariaLabel="rating" />
+		<div [(auRating)]="rating" #rat (auHover)="hovered = $event" (auLeave)="left = $event" auAriaLabel="rating" (click)="log(rat)"></div>
```
Comment (divdavem) [inline angular/demo/src/app/samples/rating/default.route.ts:8]: Adding a `(click)` event and the `log` method should not be part of this PR.

File: angular/demo/src/app/samples/alert/config.route.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ export enum AlertStatus {
 	template: ` <button class="btn btn-primary showAlert" (click)="showAlert(alert)" type="button">Show alert</button>
 		<br />
 		<br />
-		<div class="d-flex flex-column">
+		<div class="d-flex flex-column align-items-start">
```
Comment (divdavem) [inline angular/demo/src/app/samples/alert/config.route.ts:24]: This change is not related to the topic of this PR.

File: angular/lib/src/lib/rating/rating.component.ts [hunk 4]
```diff
@@ -30,7 +29,7 @@ export class RatingStarDirective {
 }
 
 @Component({
-	selector: 'au-rating',
+	selector: 'div[auRating]',
```
Comment (divdavem) [inline angular/lib/src/lib/rating/rating.component.ts:32]:
```suggestion
	selector: '[auRating]',
```
I think we should not include the tag name in the selector, so that people can use any tag they want.

**Review by divdavem — APPROVED** [review]

**Comment by quentinderoubaix** [comment]
thanks for the review @divdavem !
